torch/csrc/jit/function_schema.h

```diff
@@ -90,7 +105,9 @@ struct FunctionSchema {
   // arguments are not checked by schema
   const bool is_vararg_;
   const bool is_varret_;
-  const bool is_mutable_;
+
+  // set of alias sets in Arguments that are written to by this op
+  const std::vector<std::string> writes_;
```

Comment:
I like the vector for 2 reasons:
1. under 100 or so elements it is faster than a hash table.
2. it gives the writes a canonical order, which is important for writing deterministic algorithms. otherwise we would still need a vector to keep the deterministic order. 

Rather than use the vector directly, we should add a function to the schema for the missing functionality like: `bool writesTo(std::string x)`